src/utils/session/index.ts

```diff
@@ -0,0 +1,40 @@
+import { MsalConfig } from "../../../generated/definitions/backend/MsalConfig";
+import { getUserAgentApplication, getUserTokenOrRedirect } from "./msal";
+
+/**
+ * The shape of configuration for the current session strategy
+ */
+export type SessionConfig = MsalConfig; // will include more kinds in the future
+
+export type LoggedUserData = {};
+
+/**
+ * Clear and invalidates current session data
+ *
+ * @param configuration
+ * @returns
+ */
+export const logout = (configuration: SessionConfig): void => {
+  sessionStorage.clear();
+  return getUserAgentApplication(configuration).logout();
+};
+
+/**
+ * Get session data or redirect to login page
+ *
+ * @param configuration
+ * @returns
+ */
+export const getSessionOrLogin = async (
+  configuration: SessionConfig
+): Promise<undefined | { token: string; userData: LoggedUserData }> => {
+  const tokenAndAccount = await getUserTokenOrRedirect(configuration);
+
+  if (tokenAndAccount) {
+    return {
+      token: tokenAndAccount.token,
+      userData: tokenAndAccount.account.idToken
+    };
+  }
+  return undefined;
```

Comment:
In my personal opinion, I prefer to return a **null** when a function needs to return an object and the value doesn't exist.

Reply:
An Option would be a better choice.

Reply:
@LoFrance I would not use `null` as it can be misleading when checking data type at runtime:
```ts
typeof undefined // "undefined"
typeof null.           // "object" ¯\_(ツ)_/¯
```
I switched to `Option` as @gunzip suggested, it's indeed a more solid solution